guacamole/src/main/frontend/src/app/history/services/guacHistory.js

```diff
@@ -38,7 +39,7 @@ angular.module('history').factory('guacHistory', ['$injector',
      * The number of entries to allow before removing old entries based on the
      * cutoff.
      */
-    var IDEAL_LENGTH = 6;
+    var IDEAL_LENGTH = preferenceService.preferences.numberOfRecentConnections || 6;
```

Comment:
Does this have any impact on the behavior of `updateThumbnail()` if the setting is changed on the fly? I think `updateThumbnail()` might need pull this value itself, instead of the value being copied within `guacHistory` as a constant.

Reply:
Okay, I've just removed this completely and am relying on the preference service.